templates/NftSaleContract/src/NftSale.cs

```diff
@@ -0,0 +1,159 @@
+        }
+        
+        // transfer nft
+        public override Empty Purchase(PurchaseInput input)
+        {
+            var price = State.NftPrice.Value;
+            // transfer token
+            State.TokenContract.TransferFrom.Send(new TransferFromInput
+            {
+                From = Context.Sender,
+                To = Context.Self,
+                Symbol = price.Symbol,
+                Amount = price.Amount
+            });
+            // transfer nft
+            State.TokenContract.Transfer.Send(new TransferInput
+            {
+                To = Context.Sender,
+                Symbol = input.Symbol,
+                Amount = input.Amount
+            });
+            
+            Context.Fire(new SaleNft
+            {
+                To = Context.Self,
+                Symbol = input.Symbol,
+                Amount = input.Amount,
+            });
+            
+            return new Empty();
+        }
+        
+        // 
+        public override Empty SetPriceAndSymbol(NftPrice input)
+        {
+            AssertIsOwner();
+            State.NftPrice.Value = input.Price;
+            return new Empty();
+        }
+        
+        public override Price GetPrice(GetSymbolPriceInput input)
+        {
+            return State.NftPrice.Value;
+        }
+        
+
+        // Withdraws a specified amount of tokens from the contract.
+        // This method can only be called by the owner of the contract.
+        // After the tokens are transferred, a WithdrawEvent is fired to notify any listeners about the withdrawal.
+        public override Empty Withdraw(Int64Value input)
+        {
+            AssertIsOwner();
+            
+            // Transfer the token from the contract to the sender
+            State.TokenContract.Transfer.Send(new TransferInput
+            {
+                To = Context.Sender,
+                Symbol = State.NftPrice.Value.Symbol,
+                Amount = input.Value
+            });
+            
+            // Emit an event to notify listeners about the withdrawal
+            Context.Fire(new WithdrawEvent
+            {
+                Amount = input.Value,
+                From = Context.Self,
+                To = State.Owner.Value
+            });
+            
+            return new Empty();
+        }
+        
+        // Deposits a specified amount of tokens into the contract.
+        // This method can only be called by the owner of the contract.
+        // After the tokens are transferred, a DepositEvent is fired to notify any listeners about the deposit.
+        public override Empty Deposit(Int64Value input)
+        {
+            AssertIsOwner();
+            
+            // Transfer the token from the sender to the contract
+            State.TokenContract.TransferFrom.Send(new TransferFromInput
+            {
+                From = Context.Sender,
+                To = Context.Self,
+                Symbol = State.NftPrice.Value.Symbol,
```

Comment:
There should be a input for deposit indicating what nft symbol he wants to transfer into the contract for sale